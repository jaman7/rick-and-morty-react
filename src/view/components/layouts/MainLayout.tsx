import { Outlet } from 'react-router-dom';
import Header from '../Header/Header';

const MainLayout: React.FC = () => {
  return (
    <div className="layout">
      <Header />

      <main className="main" role="main" aria-label="Main content">
        <Outlet />
      </main>
    </div>
  );
};

export default MainLayout;
