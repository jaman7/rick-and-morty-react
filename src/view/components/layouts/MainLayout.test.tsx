import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import MainLayout from './MainLayout';

const renderWithOutlet = (component: React.ReactNode, path = '/') => {
  return render(
    <MemoryRouter initialEntries={[path]}>
      <Routes>
        <Route element={<MainLayout />}>
          <Route index element={component} />
        </Route>
      </Routes>
    </MemoryRouter>
  );
};

describe('MainLayout', () => {
  it('renders Header and main content from Outlet', () => {
    renderWithOutlet(<div data-testid="mock-outlet">Hello from page</div>);

    expect(screen.getByRole('banner')).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /Rick and Morty Characters/i })).toBeInTheDocument();

    expect(screen.getByRole('main')).toBeInTheDocument();
    expect(screen.getByTestId('mock-outlet')).toHaveTextContent('Hello from page');
  });

  it('matches snapshot', () => {
    const { container } = renderWithOutlet(<div>Hello snapshot</div>);
    expect(container).toMatchSnapshot();
  });
});
