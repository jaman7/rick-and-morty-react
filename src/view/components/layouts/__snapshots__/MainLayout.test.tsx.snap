// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainLayout > matches snapshot 1`] = `
<div>
  <div
    class="layout"
  >
    <header
      class="header"
      role="banner"
    >
      <h2
        class="header-title"
      >
        <a
          aria-current="page"
          class="active"
          data-discover="true"
          href="/"
        >
          Rick and Morty Characters
        </a>
      </h2>
      <button
        aria-controls="main-navigation"
        aria-expanded="false"
        aria-label="Open Menu"
        class="button-component primary text-sm px-3 py-2 burger-icon"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </button>
      <nav
        aria-label="Main navigation"
        class="navbar"
        id="main-navigation"
        role="navigation"
      >
        <ul
          class="menu"
        >
          <li>
            <a
              aria-current="page"
              class="item active"
              data-discover="true"
              href="/"
            >
              <span>
                Characters
              </span>
            </a>
          </li>
          <li>
            <a
              class="item"
              data-discover="true"
              href="/episodes"
            >
              <span>
                Episodes
              </span>
            </a>
          </li>
          <li>
            <a
              class="item"
              data-discover="true"
              href="/location"
            >
              <span>
                Location
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main
      aria-label="Main content"
      class="main"
      role="main"
    >
      <div>
        Hello snapshot
      </div>
    </main>
  </div>
</div>
`;
